ROSA TestROSARoleConfigReconcileExist is flaky

/kind bug
The unit test for ROSARoleConfig TestROSARoleConfigReconcileExist is flaky .


Logs as below

```
=== RUN   TestROSARoleConfigReconcileExist
    rosaroleconfig_controller_test.go:610: 
        Expected
            <string>: 
        to equal
            <string>: arn:aws:iam::123456789012:role/test-HCP-ROSA-Installer-Role
--- FAIL: TestROSARoleConfigReconcileExist (0.21s)
=== RUN   TestROSARoleConfigReconcileDelete
  GHTTP Received Request: GET - /api/clusters_mgmt/v1/clusters?page=1&search=aws.sts.operator_iam_roles.role_arn+like+%27%25%2Ftest-%25%27
--- PASS: TestROSARoleConfigReconcileDelete (0.36s)
FAIL
E1114 13:12:37.289207   49023 certwatcher.go:232] "error re-watching file" err="no such file or directory" logger="controller-runtime.certwatcher"
E1114 13:12:37.289309   49023 certwatcher.go:240] "error re-reading certificate" err="open /tmp/envtest-serving-certs-2124169368/tls.crt: no such file or directory" logger="controller-runtime.certwatcher"
E1114 13:12:37.289334   49023 certwatcher.go:232] "error re-watching file" err="no such file or directory" logger="controller-runtime.certwatcher"
E1114 13:12:37.289390   49023 certwatcher.go:240] "error re-reading certificate" err="open /tmp/envtest-serving-certs-2124169368/tls.crt: no such file or directory" logger="controller-runtime.certwatcher"
E1114 13:12:37.289419   49023 certwatcher.go:232] "error re-watching file" err="no such file or directory" logger="controller-runtime.certwatcher"
E1114 13:12:37.289469   49023 certwatcher.go:240] "error re-reading certificate" err="open /tmp/envtest-serving-certs-2124169368/tls.crt: no such file or directory" logger="controller-runtime.certwatcher"
FAIL	sigs.k8s.io/cluster-api-provider-aws/v2/exp/controllers	31.260s
```

https://prow.k8s.io/view/gs/kubernetes-ci-logs/pr-logs/pull/kubernetes-sigs_cluster-api-provider-aws/5739/pull-cluster-api-provider-aws-test/1989318443119153152#1:build-log.txt%3A4549-4555


**What steps did you take and what happened:**
[A clear and concise description of what the bug is.]


**What did you expect to happen:**


**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]


**Environment:**

- Cluster-api-provider-aws version: 
- Kubernetes version: (use `kubectl version`): 
- OS (e.g. from `/etc/os-release`):

--- a/exp/controllers/rosaroleconfig_controller_test.go
+++ b/exp/controllers/rosaroleconfig_controller_test.go
@@ -366,35 +366,34 @@ func TestROSARoleConfigReconcileCreate(t *testing.T) {
 		Client:  testEnv.Client,
 		Runtime: r,
 	}
-
-	// Call the Reconcile function
 	req := ctrl.Request{}
 	req.NamespacedName = types.NamespacedName{Name: rosaRoleConfig.Name, Namespace: rosaRoleConfig.Namespace}
-	_, errReconcile := reconciler.Reconcile(ctx, req)
 
-	// Assertions - expect the installer role empty error since AccountRolesRef is not populated yet
-	g.Expect(errReconcile).ToNot(HaveOccurred())
+	g.Eventually(func(g Gomega) {
+		// Call the Reconcile function
+		_, errReconcile := reconciler.Reconcile(ctx, req)
 
-	// Sleep to ensure the status is updated
-	time.Sleep(100 * time.Millisecond)
+		// Assertions - expect the installer role empty error since AccountRolesRef is not populated yet
+		g.Expect(errReconcile).ToNot(HaveOccurred())
 
-	// Check the status of the ROSARoleConfig resource
-	updatedRoleConfig := &expinfrav1.ROSARoleConfig{}
-	err = reconciler.Client.Get(ctx, req.NamespacedName, updatedRoleConfig)
-	g.Expect(err).ToNot(HaveOccurred())
+		// Check the status of the ROSARoleConfig resource
+		updatedRoleConfig := &expinfrav1.ROSARoleConfig{}
+		err = reconciler.Client.Get(ctx, req.NamespacedName, updatedRoleConfig)
+		g.Expect(err).ToNot(HaveOccurred())
 
-	// We expect only oidcID to be set with first reconcile happen, Account roles and Operator roles should be empty
-	g.Expect(updatedRoleConfig.Status.OIDCID).To(Equal("test-oidc-id-created"))
-	g.Expect(updatedRoleConfig.Status.AccountRolesRef).To(Equal(expinfrav1.AccountRolesRef{}))
-	g.Expect(updatedRoleConfig.Status.OperatorRolesRef).To(Equal(rosacontrolplanev1.AWSRolesRef{}))
+		// We expect only oidcID to be set with first reconcile happen, Account roles and Operator roles should be empty
+		g.Expect(updatedRoleConfig.Status.OIDCID).To(Equal("test-oidc-id-created"))
+		g.Expect(updatedRoleConfig.Status.AccountRolesRef).To(Equal(expinfrav1.AccountRolesRef{}))
+		g.Expect(updatedRoleConfig.Status.OperatorRolesRef).To(Equal(rosacontrolplanev1.AWSRolesRef{}))
 
-	// Ready condition should be false.
-	for _, condition := range updatedRoleConfig.Status.Conditions {
-		if condition.Type == expinfrav1.RosaRoleConfigReadyCondition {
-			g.Expect(condition.Status).To(Equal(corev1.ConditionFalse))
-			break
+		// Ready condition should be false.
+		for _, condition := range updatedRoleConfig.Status.Conditions {
+			if condition.Type == expinfrav1.RosaRoleConfigReadyCondition {
+				g.Expect(condition.Status).To(Equal(corev1.ConditionFalse))
+				break
+			}
 		}
-	}
+	}).WithTimeout(30 * time.Second).Should(Succeed())
 }
 
 func TestROSARoleConfigReconcileExist(t *testing.T) {
@@ -557,12 +556,12 @@ func TestROSARoleConfigReconcileExist(t *testing.T) {
 	)
 
 	// Create CRs with unique names to avoid conflicts
-	ns, err := testEnv.CreateNamespace(ctx, fmt.Sprintf("test-namespace-all-existing-%s", testID))
+	ns, err := testEnv.CreateNamespace(ctx, fmt.Sprintf("test-namespace-existing-%s", testID))
 	g.Expect(err).ToNot(HaveOccurred())
 
 	rosaRoleConfig := &expinfrav1.ROSARoleConfig{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:       fmt.Sprintf("test-rosa-role-all-existing-%s", testID),
+			Name:       fmt.Sprintf("test-rosarole-existing-%s", testID),
 			Namespace:  ns.Name,
 			Finalizers: []string{expinfrav1.RosaRoleConfigFinalizer},
 		},
@@ -590,19 +589,18 @@ func TestROSARoleConfigReconcileExist(t *testing.T) {
 		Runtime: r,
 	}
 
-	// Call the Reconcile function
 	req := ctrl.Request{}
 	req.NamespacedName = types.NamespacedName{Name: rosaRoleConfig.Name, Namespace: rosaRoleConfig.Namespace}
-	_, errReconcile := reconciler.Reconcile(ctx, req)
-
-	// Assertions - since all resources exist, reconciliation should succeed
-	g.Expect(errReconcile).ToNot(HaveOccurred())
-
-	var updatedRoleConfig *expinfrav1.ROSARoleConfig
 
 	g.Eventually(func(g Gomega) {
+		// Call the Reconcile function
+		_, errReconcile := reconciler.Reconcile(ctx, req)
+
+		// Assertions - since all resources exist, reconciliation should succeed
+		g.Expect(errReconcile).ToNot(HaveOccurred())
+
 		// Check the status of the ROSARoleConfig resource
-		updatedRoleConfig = &expinfrav1.ROSARoleConfig{}
+		updatedRoleConfig := &expinfrav1.ROSARoleConfig{}
 		g.Expect(reconciler.Client.Get(ctx, req.NamespacedName, updatedRoleConfig)).ToNot(HaveOccurred())
 
 		// Verify that all existing account roles are preserved
@@ -623,15 +621,13 @@ func TestROSARoleConfigReconcileExist(t *testing.T) {
 		g.Expect(updatedRoleConfig.Status.OperatorRolesRef.NodePoolManagementARN).To(Equal("arn:aws:iam::123456789012:role/test-kube-system-capa-controller-manager"))
 		g.Expect(updatedRoleConfig.Status.OperatorRolesRef.ControlPlaneOperatorARN).To(Equal("arn:aws:iam::123456789012:role/test-kube-system-control-plane-operator"))
 		g.Expect(updatedRoleConfig.Status.OperatorRolesRef.KMSProviderARN).To(Equal("arn:aws:iam::123456789012:role/test-kube-system-kms-provider"))
-	}).WithTimeout(30 * time.Second).Should(Succeed())
 
-	// Should have a condition indicating success - expect Ready condition to be True
-	g.Eventually(func(g Gomega) {
+		// Should have a condition indicating success - expect Ready condition to be True
 		readyCondition := v1beta1conditions.Get(updatedRoleConfig, expinfrav1.RosaRoleConfigReadyCondition)
 		g.Expect(readyCondition).ToNot(BeNil())
 		g.Expect(readyCondition.Status).To(Equal(corev1.ConditionTrue))
 		g.Expect(readyCondition.Reason).To(Equal(expinfrav1.RosaRoleConfigCreatedReason))
-	}).Should(Succeed())
+	}).WithTimeout(30 * time.Second).WithPolling(500 * time.Millisecond).Should(Succeed())
 }
 
 func TestROSARoleConfigReconcileDelete(t *testing.T) {

